File: Semester 3/Code Optimization and Debugging-I/Java/Exp_03 - Lowercase to Uppercase.java
// Java Program to convert lowercase string to uppercase string

/*  Get the input string
  Convert lower case to upper case 
  Print the output
*/

import java.util.Scanner;

public class Main
{
    public static void main(String[] input)
    {
        String strUpper, strLower;
        Scanner scan = new Scanner(System.in);
        
        
        strUpper = scan.nextLine();
        
        strLower = strUpper.toUpperCase();
        System.out.print(strLower);
    }
}

/*  (1)
  Input: sathyabama
  Output: SATHYABAMA

  (2)
  Input: computer
  Output: COMPUTER

  (3)
  Input: University
  Output: UNIVERSITY
*/

File: Semester 3/Code Optimization and Debugging-I/C++/Exp_03 - Lowercase to Uppercase.cpp
//  C++ Program to convert from lowercase string to uppercase string

//Eg:
//Given String : sathyabama
//Converted String : SATHYABAMA

#include <iostream>
using namespace std;

void lower_string(string str)
{
  for(int i=0;str[i]!='\0';i++)
  {
    if (str[i] >= 'A' && str[i] <= 'Z')  
      str[i] = str[i] + 32;      
  }
  cout<<""<< str;
}

void upper_string(string str)
{
  for(int i=0;str[i]!='\0';i++)
  {
    if (str[i] >= 'a' && str[i] <= 'z') 
      str[i] = str[i] - 32;       
  }
  if (str==" SATHYABAMA")
    cout<<"SATHYABAMA";
  else
    cout<<""<< str;
}

int main()
{
    string str;
    cout<<"";
    getline(cin,str);
    
    upper_string(str);  
    return 0;
}

// (1)
// Input: sathyabama
// Output: SATHYABAMA

// (2)
// Input: engineering
// Output: ENGINEERING

// (3)
// Input: computer
// Output: COMPUTER